app/src/main/java/io/gnosis/safe/Tracker.kt

```diff
@@ -0,0 +1,60 @@
+package io.gnosis.safe
+
+import android.content.Context
+import android.os.Bundle
+import com.google.firebase.analytics.FirebaseAnalytics
+import com.google.firebase.crashlytics.FirebaseCrashlytics
+import io.gnosis.safe.di.ApplicationContext
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class Tracker @Inject constructor(@ApplicationContext context: Context) {
+
+    private val firebaseAnalytics: FirebaseAnalytics = FirebaseAnalytics.getInstance(context)
+
```

Comment:
Singleton implementation suggestion part1:
```suggestion
init {
 this.tracker = this
}
```